src/Avalonia.Styling/Controls/ResourceProviderExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Text;
+
+namespace Avalonia.Controls
+{
+    public static class ResourceProviderExtensions
+    {
+        /// <summary>
+        /// Finds the specified resource by searching up the logical tree and then global styles.
+        /// </summary>
+        /// <param name="control">The control.</param>
+        /// <param name="key">The resource key.</param>
+        /// <returns>The resource, or <see cref="AvaloniaProperty.UnsetValue"/> if not found.</returns>
+        public static object FindResource(this IResourceNode control, string key)
```

Comment:
Is it possible to change this to `bool TryFindResource(this IResourceNode control, string key, out object value)` and not use `UnsetValue`? The latter was a source of some bugs in WPF, IIRC.

Reply:
Yeah, I was considering that but decided to go with the WPF API. I do agree that a `TryGetResource` API would be better though.

Reply:
I added `TryFindResource` but for some reason it's not showing up on GH, it's definitely there in the branch though!